src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs

```diff
@@ -66,6 +66,77 @@ public static bool IsSigned(BinaryReader reader)
             return false;
         }
 
+        /// <summary>
+        /// Opens a read-only stream for the package signature file.
+        /// </summary>
+        /// <remarks>Callers should first verify that a package is signed before calling this method.</remarks>
+        /// <param name="reader">A binary reader for a signed package.</param>
+        /// <returns>A readable stream.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="reader" /> is <c>null</c>.</exception>
+        /// <exception cref="SignatureException">Thrown if a package signature file is invalid or missing.</exception>
+        public static Stream OpenPackageSignatureFileStream(BinaryReader reader)
+        {
+            if (reader == null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            SignedPackageArchiveMetadata metadata;
+            CentralDirectoryHeaderMetadata signatureCentralDirectoryHeader;
+
+            ReadSignedPackage(reader, out metadata, out signatureCentralDirectoryHeader);
+
+            return GetPackageSignatureFile(reader, signatureCentralDirectoryHeader);
+        }
+
+        private static Stream GetPackageSignatureFile(
+            BinaryReader reader,
+            CentralDirectoryHeaderMetadata signatureCentralDirectoryHeader)
+        {
+            var localFileHeader = ReadPackageSignatureFileLocalFileHeader(reader, signatureCentralDirectoryHeader);
+            var offsetToData = signatureCentralDirectoryHeader.OffsetToFileHeader +
+                LocalFileHeader.SizeInBytesOfFixedLengthFields +
+                localFileHeader.FileNameLength +
+                localFileHeader.ExtraFieldLength;
+
+            var buffer = new byte[localFileHeader.UncompressedSize];
+
+            reader.BaseStream.Seek(offsetToData, SeekOrigin.Begin);
+            reader.BaseStream.Read(buffer, offset: 0, count: buffer.Length);
+
+            return new MemoryStream(buffer, writable: false);
+        }
+
+        private static void ReadSignedPackage(
+            BinaryReader reader,
+            out SignedPackageArchiveMetadata metadata,
+            out CentralDirectoryHeaderMetadata signatureCentralDirectoryHeader)
+        {
+            metadata = SignedPackageArchiveIOUtility.ReadSignedArchiveMetadata(reader);
+
+            SignedPackageArchiveIOUtility.AssertExactlyOnePrimarySignatureAndUpdateMetadata(metadata);
```

Comment:
shouldn't ReadSignedArchiveMetadata function also be doing the assertion? If i was the using the ReadSignedArchiveMetadata, it would be hard for me to figure out that i do need to follow that function call with AssertExactlyOnePrimarySignatureAndUpdateMetadata .